src/ncp/ncp_spi.cpp

```diff
@@ -53,6 +53,8 @@
 namespace ot {
 namespace Ncp {
 
+#if OPENTHREAD_ENABLE_SPINEL_VENDOR_SUPPORT == 0
+
```

Comment:
How is this intended to work?.... `NcpSpi`/`NcpUart` inherit from `NcpBase`... 
For vendor support, should the new vendor class be a subclass of `NcpBase` or a subclass of `NcpSpi`/`NcpUart`?

Reply:
The new class must subclass from ncpSpi or ncpUart as appropriate.  It would define its own members and would implement otNcpInit(...).  Essentially it would replace the code found in ncp_spi.cpp and ncp_uart.cpp which does the same.